sdk/tests/types/slot.rs

```diff
@@ -1,29 +1,40 @@
 // Copyright 2023 IOTA Stiftung
 // SPDX-License-Identifier: Apache-2.0
 
-use core::str::FromStr;
-
-use iota_sdk::types::block::{
-    slot::{RootsId, SlotCommitment, SlotCommitmentId, SlotIndex},
-    PROTOCOL_VERSION,
-};
+use iota_sdk::types::block::slot::SlotCommitment;
+use packable::PackableExt;
 
 #[test]
 fn slot_commitment_id() {
-    let commitment = SlotCommitment::new(
-        PROTOCOL_VERSION,
-        SlotIndex::new(10),
-        SlotCommitmentId::from_str(
-            "0x20e07a0ea344707d69a08b90be7ad14eec8326cf2b8b86c8ec23720fab8dcf8ec43a30e4a8cc3f1f",
-        )
-        .unwrap(),
-        RootsId::from_str("0xcf077d276686ba64c0404b9eb2d15556782113c5a1985f262b70f9964d3bbd7f").unwrap(),
-        5,
-        10,
+    // Test from https://github.com/iotaledger/tips-draft/blob/tip46/tips/TIP-0046/tip-0046.md#slot-commitment-id-1
+    let slot_commitment_json = r#"
+    {
+        "version":3,
+        "index":"10",
+        "previousCommitmentId":"0x4b024b3e47280d05272a7d136f0c464e4e136b734e6c427749413e286162077560652c007e37241a",
+        "rootsId":"0x75614402763f5f045c040334631b791b4d755d626d504b134a505c001c516549",
+        "cumulativeWeight":"100",
+        "referenceManaCost":"6000"
+      }
+      "#;
+
+    let slot_commitment = serde_json::from_str::<SlotCommitment>(slot_commitment_json).unwrap();
+    let slot_commitment_bytes = slot_commitment.pack_to_vec();
+
+    assert_eq!(
+        slot_commitment_bytes,
+        [
+            3, 10, 0, 0, 0, 0, 0, 0, 0, 75, 2, 75, 62, 71, 40, 13, 5, 39, 42, 125, 19, 111, 12, 70, 78, 78, 19, 107,
```

Comment:
Use the hex repr?

Reply:
I wouldn't do that tbh, it's like that in the TIP, I'm copy pasting it